### PR TITLE
Bug 1839522 – added splash screen compat implementation, tied extra duration to the experiment

### PR DESCRIPTION
**Pull request by mavduevskiy**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1839522

### CODE REVIEW

File: fenix/app/src/main/res/drawable-v23/splash_screen.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item
+        android:drawable="@mipmap/ic_launcher_round"
+        android:width="160dp"
+        android:height="160dp"
```
Comment (mavduevskiy) [inline fenix/app/src/main/res/drawable-v23/splash_screen.xml:9]: dimensions from the splash screen [doc](https://developer.android.com/develop/ui/views/launch/splash-screen#dimensions)

**Review by titooan — APPROVED** [review]
LGTM! Nice job! 👏

File: fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt [hunk 2]
```diff
@@ -411,6 +416,24 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         }
     }
 
+    private fun maybeShowSplashScreen() {
+        if (components.settings.isFirstSplashScreenShown) {
+            return
+        } else {
+            components.settings.isFirstSplashScreenShown = true
+            // Splash screen compat fails to draw icons on earlier versions.
+            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.M) return
```
Comment (jonalmeida) [inline fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt:425]: Please use braces here. The linter should have complained here.

File: fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt [hunk 3]
```diff
@@ -411,6 +416,24 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
...
+        }
+
+        if (FxNimbus.features.splashScreen.value().enabled) {
+            val splashScreen = installSplashScreen()
+            var keepSplashOnScreen = true
+            val delay = FxNimbus.features.splashScreen.value().maximumDurationMs.toLong()
+            splashScreen.setKeepOnScreenCondition { keepSplashOnScreen }
+            Handler(Looper.getMainLooper()).postDelayed({ keepSplashOnScreen = false }, delay)
```
Comment (jonalmeida) [inline fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt:433]: Was there a reason to use a Handler over the `MainScope`? We don't usually use them directly.

**Review by jonalmeida — COMMENTED** [review]
Drive-by question. 🚗

File: fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt (hunk 3 above)
Comment (mavduevskiy) [reply fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt:433]: Honestly, I just didn't know we have the preferred way and used `Handler.postDelayed` after I saw 10+ usages across the project. Changed to `MainScope(delay = `👍

Comment (jonalmeida) [reply fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt]: Some times they are valid use cases, some times they are old code that we haven't touched in a long time. 🥲

**Pull request merged**